src/testdir/test_visual.vim

```diff
@@ -1630,4 +1630,56 @@ func Test_visual_substitute_visual()
   bwipe!
 endfunc
 
+func Test_visual_getregion()
+  new
+  call setline(1, ['one', 'two', 'three'])
+
+  " Visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>vl", 'tx')
+  call assert_equal(['on'], getregion('v', '.'))
+  call assert_equal(['on'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['n'], getregion('.', '.'))
+
+  " Line visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>Vll", 'tx')
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['one'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  " Block visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>ll", 'tx')
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  " Multiline with line visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>Vjj", 'tx')
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.'))
+
+  " Multiline with block visual mode
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>jj", 'tx')
+  call assert_equal(['o', 't', 't'], getregion('v', '.'))
+
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>jj$", 'tx')
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.'))
+
+  " Invalid pattern
+  call cursor(1, 1)
+  call feedkeys("\<ESC>\<C-v>jj$", 'tx')
+  call assert_fails("call getregion(1, 2)")
+  call assert_fails("call getregion('.', {})")
```

Comment:
In the assert_fails() calls, can you also include the actual error number and the message?